libsql-server/src/connection/write_proxy.rs

```diff
@@ -264,6 +233,161 @@ impl WriteProxyConnection {
     }
 }
 
+struct RemoteConnection<R = Streaming<ExecResp>> {
+    response_stream: R,
+    request_sender: mpsc::Sender<ExecReq>,
+    current_request_id: u32,
+    builder_config: QueryBuilderConfig,
+}
+
+impl RemoteConnection {
```

Comment:
Can we remove the default type and do `RemoteConnection<Streaming<ExecResp>>` instead here?